src/main/java/org/example/dao/TicketDaoImpl.java

```diff
@@ -12,12 +13,19 @@
 import java.util.ArrayList;
 import java.util.UUID;
 
-public class TicketDaoImpl implements TicketDao {
+@Component
+public class TicketDaoImpl implements TicketDAO {
+
+    private final ConnectionConfig connectionConfig;
+
+    public TicketDaoImpl(ConnectionConfig connectionConfig) {
+        this.connectionConfig = connectionConfig;
+    }
 
     public boolean save(Ticket ticket) {
         String sqlCommand = "INSERT INTO ticket_data (id, user_id, ticket_type, creation_date)" +
                 " VALUES (?, ?, ?::ticket_type, ?)";
-        try (Connection connection = ConnectionConfig.getConnection();
+        try (Connection connection = connectionConfig.getConnection();
              PreparedStatement statement = connection.prepareStatement(sqlCommand)) {
```

Comment:
you can use:
```java
PreparedStatement statement = dataSource.getConnection().prepareStatement(sqlCommand)
```
Method `getConnection()` will be already configured by Spring under the hood for specific datasource

Reply:
Deleted Connection connection = connectionConfig.getConnection();